module/userProjects/helper.js

```diff
@@ -1715,31 +1745,39 @@ module.exports = class UserProjectsHelper {
 								projectCreation.data['userRole'] = bodyData.role
 							}
 
-							// if (solutionDetails.entityType && bodyData[solutionDetails.entityType]) {
-							// 	let entityInformation = await entitiesService.entityTypeDocuments(
-							// 		{ name: bodyData[solutionDetails.entityType] },
-							// 		'all'
-							// 	)
-
-							// 	// if( !entityInformation.success ) {
-							// 	//     throw {
-							// 	//         message : CONSTANTS.apiResponses.ENTITY_NOT_FOUND,
-							// 	//         status : HTTP_STATUS_CODE.bad_request.status
-							// 	//     }
-							// 	// }
-
-							// 	// let entityDetails = await entitiesService.entityDocuments(
-							// 	//     entityInformation.data
-							// 	// );
-
-							// 	// if ( entityDetails && entityDetails.length > 0 ) {
-							// 	//     projectCreation.data["entityInformation"] = entityDetails[0];
-							// 	// }
-
-							// 	if (entityInformation.success && entityInformation.data.length > 0) {
-							// 		projectCreation.data.entityType = entityInformation.data[0]._id
-							// 	}
-							// }
+							if (solutionDetails.entityType && bodyData[solutionDetails.entityType]) {
+								// let entityInformation = await entitiesService.entityTypeDocuments(
+								// 	{ name: bodyData[solutionDetails.entityType] },
+								// 	'all'
+								// )
+
+								// if( !entityInformation.success || !entityInformation.data.length >0  ) {
+								//     throw {
+								//         message : CONSTANTS.apiResponses.ENTITY_TYPES_NOT_FOUND,
+								//         status : HTTP_STATUS_CODE.bad_request.status
+								//     }
+								// }
+
+								let entityDetails = await entitiesService.entityDocuments({
+									_id: bodyData[solutionDetails.entityType],
+									tenantId: tenantId,
+									orgIds: { $in: [orgId] },
+								})
+
+								if (!entityDetails.success || !entityDetails?.data.length > 0) {
```

Comment:
@praveenKDass optimize
const entityDetails = await entitiesService.entityDocuments({
  _id: entityId,
  tenantId,
  orgIds: { $in: [orgId] },
});

const entityData = entityDetails?.data?.[0];

if (!entityDetails.success || !entityData) {
  throw {
    message: CONSTANTS.apiResponses.ENTITY_NOT_FOUND,
    status: HTTP_STATUS_CODE.bad_request.status,
  };
}